balto-ui/src/services/ApiClient.jsx

```diff
@@ -0,0 +1,148 @@
+import axios from "axios"
+import API_BASE_URL from ".././constants"
+
+class ApiClient {
+
+    constructor(remoteHostUrl) {
+        this.remoteHostUrl = remoteHostUrl
+        this.token = null
+        this.tokenName = "balto_token"
+    }
+
+    setToken(token) {
+        this.token = token
+        localStorage.setItem(this.tokenName, token)
+    }
+
+    // general method to make an API request
+    async request({ endpoint, method = "GET", data = {} }) {
+        const url = `${this.remoteHostUrl}/${endpoint}`
+        const headers = {
+            "Content-Type": "application/json"
+        }
+
+        // if token exists, attach Authorization header
+        if (this.token) {
+            headers["Authorization"] = `Bearer ${this.token}`
+        }
+
+        // return { data, error } object
+        try {
+            const res = await axios({ url, method, data, headers })
+            return { data: res.data, error: null}
+        } catch(error) {
+            const message = error?.response?.data?.error?.message
+            return { data: null, error: message || String(error) }
+        }
+    }
+
+    // AUTH ===============================================================================
+
+    async login(credentials) {
+        return await this.request({ endpoint:'auth/login', method:'POST', data:credentials })
+    }
+
+    async signup(credentials) {
+        return await this.request({ endpoint:'auth/register', method:'POST', data:credentials })
+    }
+
+    async logout() {
+        // NOTE: could move this to auth context once that is implemented since it doesn't actually make an API call
+        this.setToken(null)
+        localStorage.removeItem(this.tokenName)
+    }
+
+    async fetchUserFromToken() {
+        return await this.request({ endpoint:'auth/me', method:'GET' })
+    }
+
+
+    // DOGS ===============================================================================
+
+    async fetchDogs() {
+        // get dogs in database
+        // NOTE: need to revisit this once we get closer towards filtering results
+        return await this.request({ endpoint:'dogs/', method:'GET' })
+    }
+
+    async fetchDogById(dogId) {
+        // get information about an individual dog using their ID
+        return await this.request({ endpoint:`dogs/${dogId}`, method: 'GET'})
```

Comment:
correct me if i'm wrong, but isn't the endpoint "endpoint: "dogs/:dogId"?

Reply:
yeah I was confused with this too, but I think we're supposed to do it this way and then we pull the :dogId param in routes/auth.js. so the router would be like this
`   
 router.get("/:dogId", async (req, res, next) => {
        try {
            const dogId = req.params.dogId 
          ...
`

That's how I did it in my lifetracker project and it worked, but I'm not sure if this is like the "proper" way or not